src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,157 @@
+import com.revrobotics.CANSparkBase.IdleMode;
+import com.revrobotics.CANSparkBase.SoftLimitDirection;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.SparkAbsoluteEncoder.Type;
+
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.RobotMap;
+import frc.robot.utils.SparkMaxUtils;
+
+public class Elevator extends SubsystemBase {
+  public enum ElevatorPosition{
+    HOME,
+    SCORE_TRAP,
+    SCORE_AMP,
+    PRE_CLIMB,
+    CLIMB,
+    POST_CLIMB
+  }
+
+  public CANSparkMax leftMotor = new CANSparkMax(RobotMap.LEFT_ELEVATOR_CAN_ID, MotorType.kBrushless);
+  public CANSparkMax rightMotor = new CANSparkMax(RobotMap.RIGHT_ELEVATOR_CAN_ID, MotorType.kBrushless);
+
+  private final RelativeEncoder leftMotorEncoder = leftMotor.getEncoder();
+  private final AbsoluteEncoder leftMotorEncoderAbs = leftMotor.getAbsoluteEncoder(Type.kDutyCycle);
+  private final RelativeEncoder rightMotorEncoder = rightMotor.getEncoder();
+  private final AbsoluteEncoder rightMotorEncoderAbs = rightMotor.getAbsoluteEncoder(Type.kDutyCycle);
+
+  private ProfiledPIDController noteScoringElevatorController = new ProfiledPIDController(ElevatorCal.NOTE_SCORING_P,
+      ElevatorCal.NOTE_SCORING_I, ElevatorCal.NOTE_SCORING_D, new TrapezoidProfile.Constraints(
+          ElevatorCal.MAX_VELOCITY_PER_SECOND,
+          ElevatorCal.MAX_ACCELERATION_PER_SECOND_SQUARED));
+  private ProfiledPIDController chainGrabberElevatorController = new ProfiledPIDController(ElevatorCal.GRABBING_P,
+      ElevatorCal.GRABBING_I, ElevatorCal.GRABBING_D, new TrapezoidProfile.Constraints(
+          ElevatorCal.MAX_VELOCITY_PER_SECOND,
+          ElevatorCal.MAX_ACCELERATION_PER_SECOND_SQUARED));
+
+  private ProfiledPIDController currentPIDController = noteScoringElevatorController;
+
+  private TreeMap<ElevatorPosition, Double> elevatorPositions;
+
+  private ElevatorPosition desiredPosition;
+
+  private boolean usingScoringPID = true;
+
+  public Elevator() {
+    SparkMaxUtils.initWithRetry(this::initSparks, ElevatorCal.MAX_INIT_RETRY_ATTEMPTS);
+    elevatorPositions = new TreeMap<ElevatorPosition, Double>();
+    elevatorPositions.put(ElevatorPosition.HOME, ElevatorCal.POSITION_HOME);
+    elevatorPositions.put(ElevatorPosition.SCORE_AMP, ElevatorCal.POSITION_SCORE_AMP);
+    elevatorPositions.put(ElevatorPosition.SCORE_TRAP, ElevatorCal.POSITION_SCORE_TRAP);
+    elevatorPositions.put(ElevatorPosition.PRE_CLIMB, ElevatorCal.POSITION_PRE_CLIMB);
+    elevatorPositions.put(ElevatorPosition.CLIMB, ElevatorCal.POSITION_CLIMB);
+    elevatorPositions.put(ElevatorPosition.POST_CLIMB, ElevatorCal.POSITION_POST_CLIMB);
```

Comment:
should the variable names for the positions include units? like "POSITION_HOME_INCHES" or "ELEVATOR_HOME_INCHES"? Last year, for example, we named one of the positions "ELEVATOR_LOW_POSITION_INCHES"